tobago-example/tobago-example-demo/src/main/webapp/content/30-concept/53-collapsible/00-collapsible-box/Collapsible_Box.test.js

```diff
@@ -15,306 +15,206 @@
   let existContent2 = content2Fn() !== null;
 
-  let TTT = new TobagoTestTool(assert);
-  TTT.action(function () {
-    hide1Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.equal(content2Fn() !== null, existContent2);
-  });
-  TTT.action(function () {
-    show2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.ok(content2Fn() !== null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => hide1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
+  test.do(() => expect(content1Fn()).toBe(null));
+  test.do(() => expect(content2Fn() !== null).toBe(existContent2));
+  test.do(() => show2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
+  test.do(() => expect(content1Fn()).toBe(null));
+  test.do(() => expect(content2Fn() !== null).toBe(true));
+  test.start();
 });
 
-QUnit.test("Full Server Request: close both boxes", function (assert) {
-  let hide1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:hide1");
-  let hide2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:hide2");
-  let content1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content1");
-  let content2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content2");
+it("Full Server Request: close both boxes", function (done) {
+  let hide1Fn = querySelectorFn("#page\\:mainForm\\:server\\:hide1");
+  let hide2Fn = querySelectorFn("#page\\:mainForm\\:server\\:hide2");
+  let content1Fn = querySelectorFn("#page\\:mainForm\\:server\\:content1");
+  let content2Fn = querySelectorFn("#page\\:mainForm\\:server\\:content2");
   let existContent2 = content2Fn() !== null;
 
-  let TTT = new TobagoTestTool(assert);
-  TTT.action(function () {
-    hide1Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.equal(content2Fn() !== null, existContent2);
-  });
-  TTT.action(function () {
-    hide2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.equal(content2Fn(), null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => hide1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
+  test.do(() => expect(content1Fn()).toBe(null));
+  test.do(() => expect(content2Fn() !== null).toBe(existContent2));
+  test.do(() => hide2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
```

Comment:
For cleaner code: instead of _content1Fn() === null_ you can write _!content1Fn()_. We also have to check content2Fn(): _test.wait(() => !content1Fn() && !content2Fn());_